webapp/jira.py

```diff
@@ -0,0 +1,276 @@
+        """
+        Creates a task or subtask in Jira.
+
+        Args:
+            summary (str): The summary of the task.
+            issue_type (int): The ID of the issue type for the task.
+            description (str): The description of the task.
+            parent (str): The key of the parent issue. If None, the task will
+                be created without a parent.
+            reporter (str): The ID of the reporter of the task.
+            due_date (datetime): The due date of the task.
+
+        Returns:
+            dict: The response from the Jira API containing information about
+                the created task.
+        """
+        if parent:
+            parent = {"key": parent}
+
+        payload = {
+            "fields": {
+                "description": {
+                    "content": [
+                        {
+                            "content": [
+                                {
+                                    "text": description,
+                                    "type": "text",
+                                }
+                            ],
+                            "type": "paragraph",
+                        }
+                    ],
+                    "type": "doc",
+                    "version": 1,
+                },
+                "summary": summary,
+                "issuetype": {"id": issue_type},
+                "labels": self.labels,
+                "reporter": {"id": reporter},
+                "duedate": due_date.date.isoformat(),
+                "parent": parent,
+                "project": {"id": "10000"},  # Hardcoded for now
+                "components": [{"id": "10000"}],  # Hardcoded for now
+            },
+            "update": {},
+        }
+        return self.__request__(
+            method="POST", url=f"{self.url}/rest/api/3/issue", data=payload
+        )
+
+    def create_issue(
+        self,
+        issue_type: int,
+        description: str,
+        reporter: str,
+        webpage_id: int,
+        due_date: datetime,
+    ):
+        """Creates a new issue in Jira.
+
+        Args:
+            issue_type (int): The type of the issue. 0 for Epic, 1 or 2 for
+                Task.
+            description (str): The description of the issue.
+            reporter (str): The ID of the reporter.
+            webpage_id (int): The ID of the webpage.
+            due_date (datetime): The due date of the issue.
+
+        Returns:
+            dict: The response from the Jira API.
+        """
+        # Get the webpage
+        try:
+            webpage = Webpage.query.filter_by(id=webpage_id).first()
+            if not webpage:
+                raise Exception
+        except Exception:
+            raise Exception(f"Webpage with ID {webpage_id} not found")
+
+        # Get the reporter ID
+        reporter = self.get_reporter_jira_id(reporter)
+
+        # Determine the correct issue type
+        if issue_type == 0:
+            parent = None
```

Comment:
if it's a copy update type, a parent epic's key should be obtained from the env variable

Reply:
the logic you implement in this block should actually be applied to types 1 and 2. So for copy updates you create a task and attach to an existing parent, while as for page refresh and new webpage types you create an epic and 3 tasks inside that epic.

Reply:
I see, let me in fact make separate globals for the issue types that indicate which one is which